packages/apollo-gateway/src/buildQueryPlan.ts

```diff
@@ -786,12 +803,23 @@ export class QueryPlanningContext {
 
       if (!(keys && keys.length > 0)) continue;
 
-      keyFields.push(
-        ...collectFields(this, possibleType, {
-          kind: Kind.SELECTION_SET,
-          selections: keys[0],
-        }),
-      );
+      if (fetchAll) {
```

Comment:
This seems like it would lead to overfetching if other services extended this type but weren't needed:

```
type User @key(fields: "id") @key(fields: "ssn") @key(fields: "group { id }") {
}
```
If I fetched a query that needed `id` and `ssn`, would `group { id }` also be loaded here?

Reply:
Added this field to the tests, but no. As I understand it, the results of the call to getProvided are to used find which fields the service is able to supply without needing to contact any other service. So this just tells the gateway that it *can* ask for those fields, not that it must.